### PR TITLE
Switch the example to use mesop UI instead of console UI in adapter docs

### LINKED ISSUE #456: Switch the example to use Mesop UI instead of Console UI in adapter docs

### PR DESCRIPTION
**Pull request by harishmohanraj**
Fixes #456 

## Type of change

- [ ] Documentation (typos, code examples, or any documentation updates)

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -1,7 +1,6 @@
 *.jpg filter=lfs diff=lfs merge=lfs -text
 *.png filter=lfs diff=lfs merge=lfs -text
 *.pdf filter=lfs diff=lfs merge=lfs -text
-*.png filter=lfs diff=lfs merge=lfs -text
```
Comment (davorrunje) [inline .gitattributes:4]: Why would you this? This line guarantees that all PNG files are added to LFS and not directly to git.

File: docs/docs/en/getting-started/index.md [hunk 2]
```diff
@@ -393,11 +393,11 @@ Once everything is set up, you can run your FastAgency application using the fol
             ```
 
         !!! danger "Currently not working on **MacOS**"
-            The above command is currently not working on **MacOS**, please use the alternative way of starting the application from below ([#362](https://github.com/airtai/fastagency/issues/362)).
+            The above command is currently not working on **MacOS**, please use the alternative way of starting the application from below ([#362](https://github.com/airtai/fastagency/issues/362){target="_blank"}).
 
     2. Using [Gunicorn](https://gunicorn.org/){target="_blank"} WSGI HTTP server:
 
-        The preferred way to run the Mesop application is using a Python WSGI HTTP server like [Gunicorn](https://gunicorn.org/){target="_blank"}. First, you need to install it using package manager such as `pip` and then run it as follows:
+        The preferred way to run the [**Mesop**](https://google.github.io/mesop/){target="_blank"} application is using a Python WSGI HTTP server like [**Gunicorn**](https://gunicorn.org/){target="_blank"}. First, you need to install it using package manager such as `pip` and then run it as follows:
```
Comment (davorrunje) [inline docs/docs/en/getting-started/index.md:400]: gunicorn or waitress, please see bellow for the problem with Windows

File: docs/docs/en/tutorial/giphy/index.md [hunk 3]
```diff
@@ -170,11 +170,11 @@ There are two options of running a Mesop application:
         ```
 
     !!! danger "Currently not working on **MacOS**"
-        The above command is currently not working on **MacOS**, please use the alternative way of starting the application from below ([#362](https://github.com/airtai/fastagency/issues/362)).
+        The above command is currently not working on **MacOS**, please use the alternative way of starting the application from below ([#362](https://github.com/airtai/fastagency/issues/362){target="_blank"}).
 
 2. Using [Gunicorn](https://gunicorn.org/){target="_blank"} WSGI HTTP server:
 
-    The preferred way to run the Mesop application is using a Python WSGI HTTP server like [Gunicorn](https://gunicorn.org/){target="_blank"}. First, you need to install it using package manager such as `pip` and then run it as follows:
+    The preferred way to run the [**Mesop**](https://google.github.io/mesop/){target="_blank"} application is using a Python WSGI HTTP server like [**Gunicorn**](https://gunicorn.org/){target="_blank"}. First, you need to install it using package manager such as `pip` and then run it as follows:
```
Comment (davorrunje) [inline docs/docs/en/tutorial/giphy/index.md:177]: gunicorn or waitress

File: docs/docs/en/user-guide/runtimes/autogen/index.md [hunk 4]
```diff
@@ -1,132 +1,189 @@
 # AutoGen in FastAgency
 
-The **AutoGen** runtime is a key component of FastAgency, empowering developers to create intelligent, multi-agent systems powered by large language models (LLMs). AutoGen allows agents to communicate, collaborate, and perform complex tasks autonomously while easily integrating with external APIs for real-time data and functionality.
+The [**autogen**](../../../api/fastagency/runtimes/autogen/autogen/AutoGenWorkflows.md) runtime is a key component of FastAgency, empowering developers to create intelligent, [**multi-agent systems**](https://microsoft.github.io/autogen/0.2/docs/Use-Cases/agent_chat){target="_blank"} powered by [**large language models (LLMs)**](https://en.wikipedia.org/wiki/Large_language_model){target="_blank"}. It allows agents to communicate, collaborate, and perform complex tasks autonomously while easily integrating with external [**Rest APIs**](https://en.wikipedia.org/wiki/REST){target="_blank"} for real-time data and functionality.
```
Comment (davorrunje) [inline docs/docs/en/user-guide/runtimes/autogen/index.md:3]: AutoGen is writen with capital A and G, please see: https://microsoft.github.io/autogen/0.2/

File: docs/docs/en/user-guide/runtimes/autogen/index.md [hunk 5]
```diff
@@ -1,132 +1,189 @@
...
 
-In this example, we will create a simple weather chatbot using **AutoGen** in FastAgency. The chatbot will enable a user to interact with a weather agent that fetches real-time weather information from an external API using OpenAPI specifications.
+In this example, we will create a simple weather [**chatbot**](https://en.wikipedia.org/wiki/Chatbot){target="_blank"} using [**autogen**](../../../api/fastagency/runtimes/autogen/autogen/AutoGenWorkflows.md) runtime in FastAgency. The chatbot will enable a user to interact with a weather agent that fetches real-time weather information from an external REST API using [**OpenAPI specification**](https://en.wikipedia.org/wiki/OpenAPI_Specification){target="_blank"}.
```
Comment (davorrunje) [inline docs/docs/en/user-guide/runtimes/autogen/index.md:5]:
```suggestion
In this example, we will create a simple weather [**chatbot**](https://en.wikipedia.org/wiki/Chatbot){target="_blank"} using [**AutoGen**](../../../api/fastagency/runtimes/autogen/autogen/AutoGenWorkflows.md) runtime in FastAgency. The chatbot will enable a user to interact with a weather agent that fetches real-time weather information from an external REST API using [**OpenAPI specification**](https://en.wikipedia.org/wiki/OpenAPI_Specification){target="_blank"}.
```

File: docs/docs/en/user-guide/runtimes/autogen/index.md (hunk 4 above)
Comment (davorrunje) [inline docs/docs/en/user-guide/runtimes/autogen/index.md:3]:
```suggestion
The [**AutoGen**](../../../api/fastagency/runtimes/autogen/autogen/AutoGenWorkflows.md) runtime is a key component of FastAgency, empowering developers to create intelligent, [**multi-agent systems**](https://microsoft.github.io/autogen/0.2/docs/Use-Cases/agent_chat){target="_blank"} powered by [**large language models (LLMs)**](https://en.wikipedia.org/wiki/Large_language_model){target="_blank"}. It allows agents to communicate, collaborate, and perform complex tasks autonomously while easily integrating with external [**Rest APIs**](https://en.wikipedia.org/wiki/REST){target="_blank"} for real-time data and functionality.
```

File: docs/docs/en/user-guide/runtimes/autogen/index.md (hunk 5 above)
Comment (davorrunje) [inline docs/docs/en/user-guide/runtimes/autogen/index.md:5]:
```suggestion
In this example, we will create a simple weather [**chatbot**](https://en.wikipedia.org/wiki/Chatbot){target="_blank"} using [**AutoGen**](../../../api/fastagency/runtimes/autogen/autogen/AutoGenWorkflows.md) runtime in FastAgency. The chatbot will enable a user to interact with a weather agent that fetches real-time weather information from an external REST API using [**OpenAPI specification**](https://en.wikipedia.org/wiki/OpenAPI_Specification){target="_blank"}.
```

File: docs/docs/en/user-guide/runtimes/autogen/index.md [hunk 6]
```diff
@@ -1,132 +1,189 @@
...
 
 ## Installation
 
-Before getting started, make sure you have installed FastAgency with support for the AutoGen runtime by running the following command:
+Before you get started, ensure that you have FastAgency installed. Run the following command:
```
Comment (davorrunje) [inline docs/docs/en/user-guide/runtimes/autogen/index.md:9]: Why did you remove "support for the AutoGen runtime"? It's an important part of the sentence.

File: docs/docs/en/user-guide/runtimes/autogen/index.md [hunk 7]
```diff
@@ -1,132 +1,189 @@
...
 
 ```bash
-pip install "fastagency[autogen,openapi]"
+pip install "fastagency[autogen,mesop,openapi]"
 ```
 
-This installation includes the AutoGen runtime, allowing you to build multi-agent workflows and integrate external APIs seamlessly.
+This command installs the FastAgency library along with the  [**autogen**](../../../api/fastagency/runtimes/autogen/autogen/AutoGenWorkflows.md) runtime and the [**mesop**](../../../api/fastagency/ui/mesop/MesopUI.md) and [**openapi**](../../../api/fastagency/api/openapi/OpenAPI.md) submodules. These components enable you to build  [**multi-agent workflows**](https://microsoft.github.io/autogen/0.2/docs/Use-Cases/agent_chat){target="_blank"} and seamlessly integrate with the external [**Rest APIs**](https://en.wikipedia.org/wiki/REST){target="_blank"}.
```
Comment (davorrunje) [inline docs/docs/en/user-guide/runtimes/autogen/index.md:15]:
```suggestion
This command installs the FastAgency library along with the  [**AutoGen**](../../../api/fastagency/runtimes/autogen/autogen/AutoGenWorkflows.md) runtime and the [**mesop**](../../../api/fastagency/ui/mesop/MesopUI.md) and [**openapi**](../../../api/fastagency/api/openapi/OpenAPI.md) submodules. These components enable you to build  [**multi-agent workflows**](https://microsoft.github.io/autogen/0.2/docs/Use-Cases/agent_chat){target="_blank"} and seamlessly integrate with the external [**Rest APIs**](https://en.wikipedia.org/wiki/REST){target="_blank"}.
```

File: docs/docs/en/user-guide/runtimes/autogen/index.md [hunk 8]
```diff
@@ -1,132 +1,189 @@
...
 
 ### Running the Application
 
-```bash
-fastagency run
-```
+!!! note
 
-Ensure you have set your OpenAI API key in the environment and that the weather API URL is accessible. The command will launch a console interface where users can input their requests and interact with the weather agent.
+    In this example, AutoGen will utilize its [**code execution**](https://microsoft.github.io/autogen/0.2/docs/tutorial/code-executors#use-code-execution-in-conversation){target="_blank"} feature to execute external [**Rest APIs**](https://en.wikipedia.org/wiki/REST){target="_blank"} and retrieve their responses. For security reasons, AutoGen defaults to running code execution inside a [**Docker**](https://www.docker.com/){target="_blank"} container. If Docker is not running on the machine where you are executing this example, you will encounter the following error:
```
Comment (davorrunje) [inline docs/docs/en/user-guide/runtimes/autogen/index.md:126]: This is not true. AutoGen will use tool calling to access remote REST API, it will not generate and execute code.

File: docs/docs/en/user-guide/runtimes/autogen/index.md [hunk 9]
```diff
@@ -1,132 +1,189 @@
...
+
+    ```cmd
```
Comment (davorrunje) [inline docs/docs/en/user-guide/runtimes/autogen/index.md:128]: you should disable code execution in your example

File: docs/docs/en/user-guide/runtimes/autogen/index.md [hunk 10]
```diff
@@ -1,132 +1,189 @@
...
+        - Make sure docker is running (advised approach for code execution)
+        - Set "use_docker": False in code_execution_config
+        - Set AUTOGEN_USE_DOCKER to "0/False/no" in your environment variables
+    ```
+
+    For more details on this and running code execution locally without Docker, follow this guide: [**Run code execution locally**](https://microsoft.github.io/autogen/blog/2024/01/23/Code-execution-in-docker/#run-code-execution-locally){target="_blank"}.
+
+There are two options of running a Mesop application:
+
+1. Using [**`fastagency`**](../../cli/index.md) command line:
```
Comment (davorrunje) [inline docs/docs/en/user-guide/runtimes/autogen/index.md:143]:
```suggestion
1. Using [**`fastagency`**](../../cli/index.md) command:
```

File: docs/docs/en/user-guide/runtimes/autogen/using_non_openai_models.md [hunk 11]
```diff
@@ -1,84 +1,85 @@
 # Using Non-OpenAI Models with FastAgency
 
-FastAgency makes it simple to work with **non-OpenAI models** through AutoGen's runtime. You can do this in a couple of ways:
+FastAgency makes it simple to work with **non-OpenAI models** through its [**autogen**](../../../api/fastagency/runtimes/autogen/autogen/AutoGenWorkflows.md) runtime. You can do this in a couple of ways:
```
Comment (davorrunje) [inline docs/docs/en/user-guide/runtimes/autogen/using_non_openai_models.md:3]: Please search and replace [**autogen**] -> [**AutoGen**]

File: docs/docs/en/user-guide/runtimes/autogen/using_non_openai_models.md [hunk 12]
```diff
@@ -1,84 +1,85 @@
...
 
-- via [proxy servers that provide an OpenAI-compatible API](https://microsoft.github.io/autogen/0.2/docs/topics/non-openai-models/about-using-nonopenai-models/#openai-compatible-api-proxy-server){target="_blank"}
-- by [using a custom model client class](https://microsoft.github.io/autogen/0.2/docs/topics/non-openai-models/about-using-nonopenai-models/#custom-model-client-class){target="_blank"}, which lets you define and load your own models.
+- Using a [**proxy servers that provide an OpenAI-compatible API**](https://microsoft.github.io/autogen/0.2/docs/topics/non-openai-models/about-using-nonopenai-models/#openai-compatible-api-proxy-server){target="_blank"} or
```
Comment (davorrunje) [inline docs/docs/en/user-guide/runtimes/autogen/using_non_openai_models.md:5]: the sentence is broken now

File: docs/docs_src/user_guide/runtimes/autogen/mesop/main.py [hunk 13]
```diff
@@ -7,7 +7,7 @@
 from fastagency import UI, FastAgency
 from fastagency.api.openapi import OpenAPI
 from fastagency.runtimes.autogen import AutoGenWorkflows
-from fastagency.ui.console import ConsoleUI
+from fastagency.ui.mesop import MesopUI
 
 llm_config = {
```
Comment (davorrunje) [inline docs/docs_src/user_guide/runtimes/autogen/mesop/main.py:12]: make sure code execution is disabled

**Review by davorrunje — CHANGES_REQUESTED** [review]
I have to leave soon, most comment apply to the rest of the files changed.

File: docs/docs/en/getting-started/index.md (hunk 2 above)
Comment (harishmohanraj) [reply docs/docs/en/getting-started/index.md:400]: We have a note below this paragraph explaining the problem in windows and the solution. We have added this common note in all required places.

File: docs/docs/en/tutorial/giphy/index.md (hunk 3 above)
Comment (harishmohanraj) [reply docs/docs/en/tutorial/giphy/index.md:177]: We have a note below this paragraph explaining the problem in windows and the solution. We have added this common note in all required places.

File: docs/docs/en/user-guide/runtimes/autogen/index.md (hunk 6 above)
Comment (harishmohanraj) [reply docs/docs/en/user-guide/runtimes/autogen/index.md:9]: I have explained that in the next paragraph, but now moved it back where it was before.

File: docs/docs/en/user-guide/runtimes/autogen/index.md (hunk 10 above)
Comment (harishmohanraj) [reply docs/docs/en/user-guide/runtimes/autogen/index.md:143]: Removed "line" from the sentence

File: docs/docs/en/user-guide/runtimes/autogen/using_non_openai_models.md (hunk 12 above)
Comment (harishmohanraj) [reply docs/docs/en/user-guide/runtimes/autogen/using_non_openai_models.md:5]: Fixed the broken sentence

File: docs/docs/en/user-guide/runtimes/autogen/index.md (hunk 4 above)
Comment (harishmohanraj) [reply docs/docs/en/user-guide/runtimes/autogen/index.md:3]: Fixed

File: docs/docs/en/user-guide/runtimes/autogen/index.md (hunk 5 above)
Comment (harishmohanraj) [reply docs/docs/en/user-guide/runtimes/autogen/index.md:5]: Fixed

File: docs/docs/en/user-guide/runtimes/autogen/index.md (hunk 4 above)
Comment (harishmohanraj) [reply docs/docs/en/user-guide/runtimes/autogen/index.md:3]: Fixed

File: docs/docs/en/user-guide/runtimes/autogen/index.md (hunk 5 above)
Comment (harishmohanraj) [reply docs/docs/en/user-guide/runtimes/autogen/index.md:5]: Fixed

File: docs/docs/en/user-guide/runtimes/autogen/index.md (hunk 7 above)
Comment (harishmohanraj) [reply docs/docs/en/user-guide/runtimes/autogen/index.md:15]: Fixed

File: docs/docs/en/user-guide/runtimes/autogen/using_non_openai_models.md (hunk 11 above)
Comment (harishmohanraj) [reply docs/docs/en/user-guide/runtimes/autogen/using_non_openai_models.md:3]: Fixed

File: docs/docs/en/user-guide/runtimes/autogen/index.md (hunk 9 above)
Comment (harishmohanraj) [reply docs/docs/en/user-guide/runtimes/autogen/index.md:128]: Disabled the code execution in the UserProxyAgent

File: docs/docs/en/user-guide/runtimes/autogen/index.md (hunk 8 above)
Comment (harishmohanraj) [reply docs/docs/en/user-guide/runtimes/autogen/index.md:126]: Removed this section from docs. It is not relevant anymore.

File: docs/docs/en/getting-started/index.md (hunk 2 above)
Comment (davorrunje) [reply docs/docs/en/getting-started/index.md:400]: I know, but why don't we say 

The preferred way to run the [**Mesop**](https://google.github.io/mesop/){target="_blank"} application is using a Python WSGI HTTP server like [**Gunicorn**](https://gunicorn.org/){target="_blank"} or [**waitress**](something){target="_blank"}.

**Review by davorrunje — APPROVED** [review]
